Lib/test/test_import/__init__.py

```diff
@@ -111,6 +111,27 @@ def test_from_import_missing_attr_path_is_canonical(self):
         self.assertIn(cm.exception.name, {'posixpath', 'ntpath'})
         self.assertIsNotNone(cm.exception)
 
+    def test_from_import_star_invalid_type(self):
+        TESTFNnoat = TESTFN[1:]
+        source = TESTFNnoat + os.extsep + "py"
```

Comment:
There are no guaranties that TESTFN without the first character is a valid identifier or that it doesn't conflict with the name of a standard module. It would be safer to use TESTFN as the name of a directory name where put the Python file with predefined name (or different names for different cases). Or even use test.support.temp_dir().